fsw/src/crypto_props.c

```diff
@@ -0,0 +1,351 @@
+/*
+ This implementation file implements functions from crypto_props.h
+ * which are required for key,value property management via linked list
+ * data structure.  
+ */
+/* 
+ * File:   crypto_props.c
+ * Author: Ary Naim (aryan.e.naim@jpl.nasa.gov)
+ *
+ * Created on November 4, 2021, 9:37 AM
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include "stdbool.h"
+#include <string.h>
+#include "crypto_props.h"
+
+crypto_config_list* p_self = NULL; 
+/*=============================================================================
+Private Declarations
+==============================================================================*/
+int crypto_config_props_set_property_value_helper(char* key, char* value, crypto_config_list** props);
+char* crypto_config_props_get_property_value_helper(char* key, crypto_config_list** props);
+int crypto_config_props_add_property_helper(char* key, char* value,crypto_config_list** pp_self); 
+int crypto_config_props_remove_property_helper(char* key,crypto_config_list** pp_self); 
+int safe_copy(char* dest, char* src);
+/*=============================================================================
+ * Implementations
+==============================================================================*/
+crypto_config_list* crypto_config_alloc(int max_key_size, int max_value_size)
+{
+    if (NULL==p_self)
+    {
+        p_self = (crypto_config_list*) calloc(1,sizeof(crypto_config_list));
+        p_self->KEY_SIZE_IN_BYTES = max_key_size; 
+        p_self->VALUE_SIZE_IN_BYTES = max_value_size; 
+        
+    }
+    return p_self; 
+}
+
+void crypto_config_free(crypto_config_list** pp_self)
+{
+    if (NULL!=pp_self && NULL!=(*pp_self))
+    {
+        crypto_config_node* p_current = (*pp_self)->p_head; 
+        crypto_config_node* p_next = NULL; 
+        //iterate and free the nodes 
+        while (NULL!=p_current)
+        {
+            p_next = p_current->p_next; 
+            free(p_current);
+            p_current = p_next; 
+           
+        }
+        //free the super structure 
+        free (*pp_self);
+    }
+}
+
+int crypto_config_props_add_property(char* key, char* value)
+{
+    int success = 0; 
```

Comment:
- Following the way CryptoLib works now, we should keep the pattern of int32 success = OS_SUCCESS throughout, instead of this components own definition of success and the platform's default 'int' instead of explicitly 'int32'

Reply:
@IbraheemYSaleh Im not sure what OS_SUCCESS returns & its not standard C. However, <stdbool.h> defines =:

```
#define bool	_Bool
#define true	1
#define false	0
```
Therefore I changed

`success = 1` to `status = true. `

I have not defined my functions for granular error codes, they are just simple fail or succeed, with error msgs printed to standard out. 

Yes I think status is a better word, so I refactored` success `to `status`. 

These new changes will in the new pull request.